src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -107,15 +111,17 @@ static void finish_frame(framer_state* st, int is_header_boundary,
       static_cast<uint8_t>(
           (is_last_in_stream ? GRPC_CHTTP2_DATA_FLAG_END_STREAM : 0) |
           (is_header_boundary ? GRPC_CHTTP2_DATA_FLAG_END_HEADERS : 0)));
-  st->stats->framing_bytes += 9;
+  st->stats->framing_bytes += DATA_FRAME_HEADER_SIZE;
   st->is_first_frame = 0;
 }
 
 /* begin a new frame: reserve off header space, remember how many bytes we'd
    output before beginning */
 static void begin_frame(framer_state* st) {
-  st->header_idx =
-      grpc_slice_buffer_add_indexed(st->output, GRPC_SLICE_MALLOC(9));
```

Comment:
instead of rolling out GRPC_SLICE_MALLOC, why not simply inline grpc_slice_malloc?

Reply:
grpc_slice_malloc() appears to be public API so I'd rather not change it.

A compromise though: adding an inlined header method somewhere that specifically constructs inlined slices. Possibly in slice_internal.h. What do you think?

Reply:
Ah I see. Let's do that in a different PR then

Reply:
Ack; in that case I'll leave this specific code as is, and submit a followup PR that makes an inlined method as described.